enhancements/baremetal/ztp-metal3.md

```diff
@@ -0,0 +1,334 @@
+  is connected. As a result, Ironic cannot disconnect the ISO from the
+  machine's BMC, nor can it update the BMC's boot device settings properly.
+
+- The *live ISO* flow disables Ironic *inspection* process and has to simulate
+  it by updating the `BareMetalHost` based on the assisted service's inspection
+  process. This creates an unnessesary coupling between the assisted service
+  and the baremetal-operator's hardware inventory format (which, in turn, is
+  derived from the Ironic's inventory format).
+
+- The *standard* flow is highly [extensible][hardware managers] via the
+  [ironic-agent-image][ironic-agent] container image. This extensibility does
+  not apply to the *live ISO* flow since it does not use the Ironic agent.
+
+- Any different path through the `BareMetalHost` state machine makes it
+  exponentially more difficult for upstream developers to reason about the
+  effects of any changes. In addition, the *live iso* flow is not tested
+  upstream by the Metal3 project.
+
+### Goals
+
+- ZTP follows the *standard* Metal3 flow instead of *live ISO*.
+- (i)PXE boot, firwmare and RAID settings work for ZTP without introducing
+  special API in Metal3 components.
+- Continue to allow non-ZTP assisted deployments without depending on Metal3.
+- Not depend on assisted components in the core metal platform
+
+### Non-Goals
+
+- Converging two agent implementations (Ironic and assisted) into a single one.
+- Supporting per-host network configuration through the BMH
+  PreprovisioningNetworkData API in the ZTP case.
+- Performing firmware config changes only after the assisted service runs host
+  validations.
+
+## Proposal
+
+- Building CoreOS images for the *standard* flow is handled by
+  [image-customization-controller][icc] (ICC). It will be updated to optionally
+  accept additional Ignition configuration via a new annotation
+  `baremetal.openshift.io/ignition-override-uri`.
+
+- Additionally, ICC will recognize the label linking `BareMetalHost`s to
+  `InfraEnv` resources from the assisted service. When such a label is present,
+  ICC will wait for the Ignition override annotation to be provided.
+  Otherwise, it will proceed with the image building.
+
+- [Ironic agent][ironic-agent] currently contains a custom deployment plugin
+  (*custom deploy method* in baremetal-operator terms, *custom deploy step* in
+  Ironic terms) that invokes `coreos-installer` for the image installation.
+  A new custom deploy method `start_assisted_install` will be added to start
+  the assisted agent at the right moment.
+
+- The assisted service (more specifically, BMAC) will stop explicitly disabling
+  *inspection* on `BareMetalHost`s it manages, will start using a different
+  process based on the  new `start_assisted_install` deploy method and stop
+  rebooting the machine in the end of the process.
+
+- BMAC will also set the Ignition URI with the assisted agent configuration on
```

Comment:
I was a bit confused when reading this paragraph. It may be worth saying that BMAC will set the `ignition-override-url` annotation that is mentioned in the first paragraph of the proposal.